docs/pages/setup/guides/terraform-provider.mdx

```diff
@@ -0,0 +1,236 @@
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: terraform-impersonator
+spec:
+  # SSH options used for user sessions 
+  options:
+    # max_session_ttl defines the TTL (time to live) of SSH certificates 
+    # issued to the users with this role.
+    max_session_ttl: 10h
+
+  # allow section declares a list of resource/verb combinations that are
+  # allowed for the users of this role. by default nothing is allowed.
+  allow:
+    impersonate:
+      users: ['terraform']
+      roles: ['terraform']
+
+  # the deny section uses the identical format as the 'allow' section.
+  # the deny rules always override allow rules.
+  deny:
+    node_labels:
+      '*': '*'
+```
+
+```bash
+tctl create terraform-impersonator.yaml
+```
+
+Assign this role to the current user. Re-login to assume the new role and try
+to issue certificate for terraform user again.
+
+</Admonition>
+
+## Step 3/4. Create Terraform configuration
+
+Create a `main.tf` terraform file:
+
+```
+terraform {
+  required_providers {
+    teleport = {
+      version = ">= (=teleport.version=)"
+      source  = "gravitational.com/teleport/teleport"
+    }
+  }
+}
+
+provider "teleport" {
+  # Update addr to point to Teleport Auth/Proxy
+  addr               = "proxy.example.com:443"
+  identity_file_path = "terraform-identity"
+}
+
+resource "teleport_role" "terraform-test" {
+  metadata {
+    name        = "terraform-test"
+    description = "Terraform test role"
+    expires     = "2022-10-12T07:20:50.52Z"
+    labels = {
+      example  = "yes"
+    }
+  }
+  
+  spec {
+    options {
+      forward_agent           = false
+      max_session_ttl         = "7m"
+      port_forwarding         = false
+      client_idle_timeout     = "1h"
+      disconnect_expired_cert = true
+      permit_x11forwarding    = false
+      request_access          = "denied"
+    }
+
+    allow {
+      logins = ["this-user-does-not-exist"]
+
+      rules {
+        resources = ["user", "role"]
+        verbs = ["list"]
+      }
+
+      request {
+        roles = ["example"]
+        claims_to_roles {
+          claim = "example"
+          value = "example"
+          roles = ["example"]
+        }
+      }
+
+      node_labels {
+         key = "example"
+         value = ["yes"]
+      }
+    }
+
+    deny {
+      logins = ["anonymous"]
+    }
+  }
+}
+
+resource "teleport_user" "terraform-test" {
+  metadata {
+    name        = "terraform-test"
+    description = "Test terraform user"
+    expires     = "2022-10-12T07:20:50.52Z"
+
+    labels = {
+      test      = "true"
+    }
+  }
+
+  spec {
+    roles = ["terraform-test"]
+  }
+}
+```
+
+<Admonition type="note">
+Update `teleport.example.com:443` with the address of your Teleport cluster.
+</Admonition>
+
+## Step 4/4. Apply the configuration
+
+Check the contents of `teleport-terraform` folder:
+
+```bash
+ls 
+main.tf  terraform-identity  terraform-impersonator.yaml  terraform.yaml
```

Comment:
Not a problem here per se - I personally like the idea of having two distinct kinds of "example" boxes:

1. Copy-pasteable instructions or commands.
2. Non copyable output messages or step confirmations (as in, "if you do so and so correctly, you'll see xyz")

Reply:
we will improve it once https://github.com/gravitational/next/issues/275 is ready